### PR TITLE
adds reshaping functions to Images and Series

### PR DESCRIPTION
**Pull request by jwittenbach**
implements #335 for `Series` and a corresponding function for `Images`

### CODE REVIEW

**Comment by jwittenbach** [comment]
@freeman-lab: I want to use this to make the handling of non-2D arrays nice and straight-forward in Thunder Factorization. Seemed general enough to include directly in Thunder. Thoughts?

**Comment by freeman-lab** [comment]
Cool, seems reasonable, is this the only case where the thunder API directly references `keys` and `values`? That's the only thing that concerns me a bit. If we've avoided that so far, might not want to introduce it just for this. Other options, off the top of my head:

- You could call both `reshape` but it's confusing if they do different things
- You could only add a `reshape` method to `series`, because for images you can always do reshaping via a `map` operation, but that doesn't work for "reshaping" the keys in `series` data

Other ideas?

**Comment by jwittenbach** [comment]
Yeah, I also thought about just calling it `reshape`, but decided it was too confusing. We do mention "values" in a couple of places (e.g. `value_shape`), but there is no equivalent for "keys", which makes things tricky.

File: thunder/series/series.py [hunk 1]
```diff
@@ -256,6 +256,21 @@ def min(self):
...
+        Parameters
+        ----------
+        shape: one or more ints
+            Shape of the new keys
+        """
+        if prod(self.baseshape) != prod(shape):
+            raise ValueError("Total number of series elements must remain unchanged")
+
+        newshape = shape + (self.shape[-1], )
+        return self._constructor(self.values.reshape(newshape)).__finalize__(self)
```
Comment (freeman-lab) [inline thunder/series/series.py]: should be keys right?

File: thunder/series/series.py [hunk 2]
```diff
@@ -256,6 +256,21 @@ def min(self):
         """
         return self._constructor(self.values.min(axis=self.baseaxes, keepdims=True))
 
+    def reshape_keys(self, *shape):
+        """
+        Reshape the keys of the time series
```
Comment (freeman-lab) [inline thunder/series/series.py]: shouldn't explicitly mention time

File: thunder/series/series.py (hunk 1 above)
Comment (freeman-lab) [inline thunder/series/series.py]: oh sorry, nevermind, this is correct, i was confused

**Comment by jwittenbach** [comment]
@freeman-lab updated things to reflect our conversation yesterday...both `Images` and `Series` now have a `reshape` function with a simpler API that just takes the new shape, and then catches invalid reshapes.

**Comment by freeman-lab** [comment]
Nice @jwittenbach , this looks great! Merging in.

**Pull request merged**